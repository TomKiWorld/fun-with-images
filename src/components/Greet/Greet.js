import React from 'react';

/**
 * Greet component
 * - Display short greet and amount of entries the user used the app
 * 
 * Required props:
 * - name => User name from state
 * - entries => Amount of times images were submitted from state
 */
const Greet = ({name, entries}) => {
  const greet = entries > 0 ? 'Welcome back' : 'Lets start';
  return (
    <section className='container pa4'>
      <div className='white f3'>
        <h1>{greet} <span className='capitalize'>{name}</span></h1>
        <p>Your current entry count is:</p>
      </div>
      <div className='white f1'>
        <p>#{entries}</p>
      </div>
    </section>
  );
}

export default Greet;
