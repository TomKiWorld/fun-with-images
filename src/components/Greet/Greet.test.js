import { shallow } from 'enzyme';
import React from 'react';
import Greet from './Greet';

describe('Greet Component', () => {
  it('Expect to render Greet Component', () => {
    expect(shallow(<Greet />)).toMatchSnapshot();
  });

  it('Expect to render Greet Component with welcome back', () => {
    expect(shallow(<Greet entries={18}/>)).toMatchSnapshot();
  });
});
