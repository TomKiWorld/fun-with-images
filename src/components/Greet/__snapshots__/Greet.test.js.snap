// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greet Component Expect to render Greet Component 1`] = `ShallowWrapper {}`;

exports[`Greet Component Expect to render Greet Component with welcome back 1`] = `ShallowWrapper {}`;
